src/pages/task/actionButtons/PreviousCommits.tsx

```diff
@@ -0,0 +1,229 @@
+import { useReducer, useEffect } from "react";
+import { useLazyQuery, useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import { Option, Select } from "@leafygreen-ui/select";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Link } from "react-router-dom";
+import { ConditionalWrapper } from "components/ConditionalWrapper";
+import { finishedTaskStatuses } from "constants/task";
+import { size } from "constants/tokens";
+import {
+  GetBaseVersionAndTaskQuery,
+  GetBaseVersionAndTaskQueryVariables,
+  GetLastMainlineCommitQuery,
+  GetLastMainlineCommitQueryVariables,
+} from "gql/generated/types";
+import {
+  GET_BASE_VERSION_AND_TASK,
+  GET_LAST_MAINLINE_COMMIT,
+} from "gql/queries";
+import { TaskStatus, CommitTask, CommitType } from "types/task";
+import { errorReporting, string } from "utils";
+import { initialState, reducer } from "./reducer";
+
+const { applyStrictRegex } = string;
+const { reportError } = errorReporting;
+
+interface PreviousCommitsProps {
+  taskId: string;
+}
+export const PreviousCommits: React.FC<PreviousCommitsProps> = ({ taskId }) => {
+  const [
+    {
+      selectState,
+      disableButton,
+      link,
+      shouldFetchLastPassing,
+      shouldFetchLastExecuted,
+      hasFetchedLastPassing,
+      hasFetchedLastExecuted,
+    },
+    dispatch,
+  ] = useReducer(reducer, initialState);
+
+  const { data: taskData } = useQuery<
+    GetBaseVersionAndTaskQuery,
+    GetBaseVersionAndTaskQueryVariables
+  >(GET_BASE_VERSION_AND_TASK, {
+    variables: { taskId },
+  });
+
+  const [fetchParentTask] = useLazyQuery<
+    GetLastMainlineCommitQuery,
+    GetLastMainlineCommitQueryVariables
+  >(GET_LAST_MAINLINE_COMMIT, {
+    onCompleted: (data) => {
+      dispatch({
+        type: "setParentTask",
+        task: getTaskFromMainlineCommitsQuery(data),
+      });
+    },
+  });
+
+  const [fetchLastPassing] = useLazyQuery<
+    GetLastMainlineCommitQuery,
+    GetLastMainlineCommitQueryVariables
+  >(GET_LAST_MAINLINE_COMMIT, {
+    onCompleted: (data) => {
+      dispatch({
+        type: "setLastPassingTask",
+        task: getTaskFromMainlineCommitsQuery(data),
+      });
+    },
+  });
+
+  const [fetchLastExecuted] = useLazyQuery<
+    GetLastMainlineCommitQuery,
+    GetLastMainlineCommitQueryVariables
+  >(GET_LAST_MAINLINE_COMMIT, {
+    onCompleted: (data) => {
+      dispatch({
+        type: "setLastExecutedTask",
+        task: getTaskFromMainlineCommitsQuery(data),
+      });
+    },
+  });
+
+  const { baseTask, versionMetadata, buildVariant, displayName } =
+    taskData?.task ?? {};
+  const { projectIdentifier, order } = versionMetadata?.baseVersion ?? {};
+  const skipOrderNumber = versionMetadata?.isPatch ? order + 1 : order;
```

Comment:
This logic seems wrong to me a patches order number is not the same as a mainline commits order number. 
Patch order numbers are only relevant to the user who created the patch. (How many patches the user created)

While a mainline commits order number is specific to the project. (How many mainline commits were created.)

Reply:
You would need to look at the base commits order number.